src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -88,14 +90,32 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
-            // TODO: Investigate why this check is needed.
-            if (Propagators.DefaultTextMapPropagator.Extract(default, request, HttpRequestMessageContextPropagation.HeaderValuesGetter) != default)
+            var context = Propagators.DefaultTextMapPropagator.Extract(default, request, HttpRequestMessageContextPropagation.HeaderValuesGetter);
+            if (context != default && request.Properties.TryGetValue(SpanAttributeConstants.PreviousTryContextKey, out var previousContext))
             {
-                // this request is already instrumented, we should back off
+                // Handling request retry or redirect.
+                var retryCount = 1;
+                if (request.Properties.TryGetValue(SpanAttributeConstants.RetryCountKey, out var previousRetryCount))
+                {
+                    retryCount = (int)previousRetryCount + 1;
+                }
+
+                // Suppressing activity started by HttpClient DiagnosticsHandler.
```

Comment:
this section is a bit unclear without some examples and explanation.
some questions:
1. Does httpClient diagnostichandler created a single Activity only, or does it create one for every retry?
2. Which one are we stopping here.
3. We are creating a new activity here - is it supposed to be the child of the original http client activity or is it expected to be its sibling (i.e same parent, likely that of the Asp.Net Core Request)
4. Why do we explicitly set Activity.Current in line 108?